src/NLog/Layouts/LayoutParser.cs

```diff
@@ -457,7 +460,10 @@ private static void SetDefaultPropertyValue(ConfigurationItemFactory configurati
             }
             else
             {
-                // TODO NLog 5.0 Should throw exception when invalid configuration (Check throwConfigExceptions)
+                if (throwConfigExceptions ?? LogManager.ThrowConfigExceptions == true)
```

Comment:
Why not: `if (throwConfigExceptions ?? LogManager.ThrowConfigExceptions ?? LogManager.ThrowExceptions)` ?

Reply:
This causes the test below to fail

https://github.com/NLog/NLog/blob/dev/tests/NLog.UnitTests/LayoutRenderers/Wrappers/WhenEmptyTests.cs#L72

Reply:
Can you show why it fails? I'm guessing that it should test that `${environment:LOG_DIR_XXX}` should not fail when `LOG_DIR_XXX` does not it exists.

Reply:
Maybe something is wrong with the literal? The test passes if I remove the value of `text` like below:

```c#
SimpleLayout l = @"${whenEmpty:whenEmpty=${literal:text=}:inner=${environment:LOG_DIR_XXX}}api.log";
```

Stack traces when adding `?? LogManager.ThrowExceptions` here or in the other method:
```
Message: 
    NLog.NLogConfigurationException : NLog.LayoutRenderers.LiteralLayoutRenderer has no default property to assign value logs}
Stack Trace: 
    LayoutParser.SetDefaultPropertyValue(ConfigurationItemFactory configurationItemFactory, LayoutRenderer layoutRenderer, String value, Nullable`1 throwConfigExceptions) line 465
    LayoutParser.ParseLayoutRenderer(ConfigurationItemFactory configurationItemFactory, SimpleStringReader stringReader, Nullable`1 throwConfigExceptions) line 419
    LayoutParser.CompileLayout(ConfigurationItemFactory configurationItemFactory, SimpleStringReader sr, Nullable`1 throwConfigExceptions, Boolean isNested, String& text) line 104
    LayoutParser.ParseLayoutRenderer(ConfigurationItemFactory configurationItemFactory, SimpleStringReader stringReader, Nullable`1 throwConfigExceptions) line 400
    LayoutParser.CompileLayout(ConfigurationItemFactory configurationItemFactory, SimpleStringReader sr, Nullable`1 throwConfigExceptions, Boolean isNested, String& text) line 104
    SimpleLayout.SetLayoutText(String value, Nullable`1 throwConfigExceptions) line 137
```

```
Message: 
    NLog.NLogConfigurationException : Unrecognized property 'inner=${environment` for ${literal} (NLog.LayoutRenderers.LiteralLayoutRenderer)
  Stack Trace: 
    LayoutParser.ParseLayoutRenderer(ConfigurationItemFactory configurationItemFactory, SimpleStringReader stringReader, Nullable`1 throwConfigExceptions) line 391
    LayoutParser.CompileLayout(ConfigurationItemFactory configurationItemFactory, SimpleStringReader sr, Nullable`1 throwConfigExceptions, Boolean isNested, String& text) line 104
    LayoutParser.ParseLayoutRenderer(ConfigurationItemFactory configurationItemFactory, SimpleStringReader stringReader, Nullable`1 throwConfigExceptions) line 400
    LayoutParser.CompileLayout(ConfigurationItemFactory configurationItemFactory, SimpleStringReader sr, Nullable`1 throwConfigExceptions, Boolean isNested, String& text) line 104
    SimpleLayout.SetLayoutText(String value, Nullable`1 throwConfigExceptions) line 137
```

Reply:
Seems that `${literal:text=}` has special powers that affects normal parsing. It seems it tries to swallow more than expected since it reaches `inner=${environment`.  See also https://github.com/NLog/NLog/wiki/Literal-Layout-Renderer  Maybe there is a unwanted feature when handling parsing of `${literal}`.

Reply:
@njqdev If you feel that crawling around in the LayoutRenderer-parser-dungeon is a bit too much, then please tell. It was a little unexpected that dragons should be lurking here.

Reply:
Sure, I can investigate it and get back to you later.

Reply:
@snakefoot it's a test for when `WrapperLayoutRendererBase.Inner` is null. Here's a memory refresher: https://github.com/NLog/NLog/pull/2908

The text not being escaped in `literal:text=c:\logs\` is what caused it to throw after my changes, see #2844. So the test was passing before because of the parsing failing, config exceptions not being thrown and a layout renderer with a null Inner being created.

I could change the test to the below, which results in a null Inner and the test passing. Or change `LogManager.ThrowExceptions` to false in the test, if that makes sense. What do you think?

```c#
        [Fact]
        public void WhenEmpty_MissingInner_ShouldNotThrow()
        {
            LogManager.ThrowExceptions = true;
            SimpleLayout l = @"${whenEmpty:whenEmpty=${literal:text=c\:\\logs\\}}api.log";
            var le = LogEventInfo.Create(LogLevel.Info, "logger", "message");
            Assert.Equal("api.log", l.Render(le));
        }
```

Reply:
Thank you for the reminder. So the unit-test was created to show that it would not explode when `Inner = null` because of faulty escaping in `${literal:text=c:\logs\}`. So actually this PR will correctly reject this faulty escaping (which is good), but then it will fail to validate that `Inner == null` will not cause issues. Maybe change the unit-test into this:

```c#
        [Fact]
        public void WhenEmpty_MissingInner_ShouldNotThrow()
        {
          using (new NoThrowNLogExceptions())
          {
              SimpleLayout l = @"${whenEmpty:whenEmpty=${literal:text=c\:\\logs\\}}api.log";
              var le = LogEventInfo.Create(LogLevel.Info, "logger", "message");
              LogManager.ThrowExceptions = true;
              Assert.Equal("api.log", l.Render(le)); // Assert that missing Inner-property will not throw
          }
        }
```

Reply:
@njqdev Sorry was a little too quick with my suggetion. Have updated my code-suggestion above, so the LayoutParser will continue to fail, so `Inner`-property remains `null`.